### PR TITLE
Nxt account lookup

### PR DESCRIPTION
**Pull request by toenu23**
Thanks for your consideration.

**What does your Instant Answer do?**
Display information about a Nxt (cryptocurrency) account (See http://nxt.org for info about Nxt)
It displays the balance of the account, the amount of coins "forged" (mined), and the amount assets in one's possession.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Makes it easier to lookup e.g. one's account balance, without having to go a block explorer or start the client.
Also makes it easy to get some quick info about someone else's account.

**What is the data source for your Instant Answer? (Provide a link if possible)**
The API of mynxt.info: https://wallet.mynxt.info/docs/
It also uses http://goqr.me/api/ to create a QR code

**Why did you choose this data source?**
It's a Nxt API run by a reliable website/company. In principle, any accessible Nxt node could be used.

**Are there any other alternative (better) data sources?**
Not that I'm aware of. Of course, DDG could run an own Nxt node, to not rely on a 3rd party service.

**What are some example queries that trigger this Instant Answer?**
Any NXT account in Reed-Solomon format, e.g.:
NXT-AR77-SW4Y-M3VA-DUSTY
NXT-W6CT-NPDH-AAQW-HWCHA

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
For anyone active in the Nxt ecosystem: crypto/blockchain enthusiasts, programmers, investors, marketplace buyers/seller etc.
It's a young, growing community.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No.

**Which existing Instant Answers will this one supersede/overlap with?**
None.

**Are you having any problems? Do you need our help with anything?**
No, making this was pretty straight forward. Thanks for the great docs.

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**
An account at mynxt.info is required. API usage is free.
goqr.me API also is free, with no account required.

**Where did you hear about DuckDuckHack? (For first time contributors)**
Don't remember. :)

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
http://i.imgur.com/IgLb43r.png
## Checklist

Please place an 'X' where appropriate.

```
[x] Added metadata and attribution information
[x] Wrote test file and added to t/ directory
[x] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[x] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[x] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [x] Google Chrome
        [x] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [x] IE 10

    - Windows XP
        [] IE 7
        [] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [x] Firefox
        [x] Native Browser
        [x] Google Chrome
        [] Opera

    - Ubuntu Linux
        [x] Chrome
        [x] Firefox
```

https://duck.co/ia/view/nxt_account

### CODE REVIEW

**Comment by moollaza** [comment]
@toenu23 thanks a lot for contributing, we really appreciate it! We'll be sure to review this shortly and give you some feedback!

**Comment by moollaza** [comment]
@mintsoft @MrChrisW @javathunderman @mattr555 -- anyone interested in giving this one a first look?

**Comment by javathunderman** [comment]
@moollaza Considering this...I'll get back to you after I finish the ISS IA.

**Comment by mintsoft** [comment]
@toenu23 Sorry for the delay! This appears to have slipped through the cracks, at first glance however it looks good to me!

**Comment by mintsoft** [comment]
@toenu23 Can you attach an image of what the Spice looks like for the design guys?

**Comment by javathunderman** [comment]
Thanks @mintsoft. Currently busy with #1559

File: share/spice/nxt_account/nxt_account.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    env.ddg_spice_nxt_account = function(api_result) {
+
+        if (!api_result) {
+            return Spice.failed('nxt_account');
+        }
+
+        Spice.add({
+            id: "nxt_account",
+            name: "Nxt Blockchain",
```
Comment (jagtalon) [inline share/spice/nxt_account/nxt_account.js]: Maybe it should just be "Software"? CC @moollaza

Comment (moollaza) [inline share/spice/nxt_account/nxt_account.js]: I don't think Software is quite right. We use "Bitcoin Wallet" for the related Bitcoin IA, I think "Nxt Wallet", or switching both to "Currency" seems OK.

**Comment by toenu23** [comment]
@mintsoft here's a screenshot of the Spice:

image

**Comment by jagtalon** [comment]
@toenu23 hey! So sorry for the delay on this. Checking it out!

**Comment by toenu23** [comment]
No problem @jagtalon take your time.
If this one goes through, I may have some more Nxt-related Spices, e.g. for its asset exchange and alias system features.

File: share/spice/nxt_account/nxt_account.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+(function (env) {
+    "use strict";
+
+    var formatNxt = function(amount) {
+        return (amount / 100000000)+" NXT"
+    };
+
+    env.ddg_spice_nxt_account = function(api_result) {
+
+        if (!api_result) {
```
Comment (jagtalon) [inline share/spice/nxt_account/nxt_account.js]: Return immediately when "status" is equal to "error" 

screen shot 2015-04-10 at 8 48 15 am

**Comment by jagtalon** [comment]
@toenu23 It's not getting the length property so the IA doesn't show:

screen shot 2015-04-10 at 8 54 23 am

**Comment by moollaza** [comment]
ping! @toenu23 just checking if you're still in interested in finishing this up? There's no rush, but we'd love to see it go live!

**Comment by toenu23** [comment]
@moollaza @jagtalon Please see the last two commits, they correct the issues.
Sorry, must have forgotten to leave a comment to notify you.

**Comment by jagtalon** [comment]
@toenu23 thanks! Checking it out!

**Comment by jagtalon** [comment]
@toenu23 I'm getting results and `NaN`s for invalid accounts. It shouldn't.

screen shot 2015-05-05 at 2 01 47 pm

**Comment by toenu23** [comment]
@jagtalon fixed it. Sorry for the delay

**Comment by abeyang** [comment]
Hi @toenu23 , it's looking good. But can we lose the zebra stripes (by getting ride of the row highlight class), so it'll look more consistent with our list templates? Should look like this in the end:

screen shot 2015-05-11 at 11 59 00 am

**Comment by toenu23** [comment]
@abeyang : done!

**Comment by abeyang** [comment]
Looks good! :+1: /cc @moollaza @jagtalon

**Comment by moollaza** [comment]
LGTM :+1:

**Comment by moollaza** [comment]
@toenu23 Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

**Comment by moollaza** [comment]
@toenu23 congrats! It's live: https://duckduckgo.com/?q=nxt-fz4l-u4ma-mam2-d3u36&ia=nxtblockchain

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**